cdap-elastic/src/test/java/co/cask/cdap/metadata/elastic/ElasticsearchMetadataStorageTest.java

```diff
@@ -303,4 +312,40 @@ public void testFiltering() {
   protected List<String> getAdditionalTTLQueries() {
     return ImmutableList.of("ttl:0003600", "TtL:03600", "TtL:03600.00");
   }
+
+  @Test
+  public void testScrollTimeout() throws IOException, InterruptedException {
+    MetadataStorage mds = getMetadataStorage();
+
+    List<MetadataRecord> records = IntStream.range(0, 20).boxed().map(i -> new MetadataRecord(
+      MetadataEntity.ofDataset("ns" + i, "ds" + i),
+      new Metadata(MetadataScope.USER, tags("tag", "t" + i), props("p", "v" + i)))).collect(Collectors.toList());
+    mds.batch(records.stream().map(r -> new Update(r.getEntity(), r.getMetadata())).collect(Collectors.toList()));
+
+    SearchRequest request = SearchRequest.of("t*").setCursorRequested(true).setLimit(5).build();
+    SearchResponse response = mds.search(request);
+    Assert.assertEquals(5, response.getResults().size());
+    Assert.assertNotNull(response.getCursor());
+
+    SearchRequest request2 = SearchRequest.of("t*").setCursor(response.getCursor()).build();
+    SearchResponse response2 = mds.search(request2);
+    Assert.assertEquals(5, response.getResults().size());
+
+    // sleep 1 sec longer than the configured scroll timeout to invalidate cursor
+    TimeUnit.SECONDS.sleep(3);
```

Comment:
The next test with invalid cursor tests the code path for handling `ElasticsearchStatusException`, do we need the sleep to timeout cursor test too?

Reply:
a time-out is different from an invalid cursor. Because Elastic's documentation is not correct, I didn't want to rely on my derived knowledge that they both cause the same error. Strictly speaking, the timeout is handled by a different code path (the one that handles a response which indicates the timeout, instead of handling an exception).